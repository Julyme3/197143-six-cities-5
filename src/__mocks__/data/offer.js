/* eslint-disable */
export const offersFromAPi = [
  {
    bedrooms: 3,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
    host: {
      id: 25,
      name: `Angelina`,
      is_pro: true,
      avatar_url: `img/avatar-angelina.jpg`,
    },
    id: 1,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.957361,
      longitude: 6.9509739999999995,
      zoom: 16,
    },
    max_adults: 4,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    price: 493,
    rating: 3.8,
    title: `The house among olive`,
    type: `house`,
  },
  {
    bedrooms: 3,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
    host: {
      id: 25,
      name: `Angelina`,
      is_pro: true,
      avatar_url: `img/avatar-angelina.jpg`,
    },
    id: 2,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.957361,
      longitude: 6.9509739999999995,
      zoom: 16,
    },
    max_adults: 4,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    price: 493,
    rating: 3.8,
    title: `The house among olive`,
    type: `house`,
  },
];

export const offersAdapteToClient = [
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 1,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  },
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 2,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  },
];

export const offerFromApi = {
  bedrooms: 3,
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13,
    },
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
  host: {
    id: 25,
    name: `Angelina`,
    is_pro: true,
    avatar_url: `img/avatar-angelina.jpg`,
  },
  id: 1,
  images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
  is_favorite: false,
  is_premium: false,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16,
  },
  max_adults: 4,
  preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
  price: 493,
  rating: 3.8,
  title: `The house among olive`,
  type: `house`,
};

export const fullOfferAdaptedToClient = {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 1,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  };

export const commentsFromApi = [
  {
    comment: "The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.",
    date: "2020-11-05T13:38:44.753Z",
    id: 1,
    rating: 4,
    user: {
      avatar_url: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg",
      id: 16,
      is_pro: true,
      name: "Mollie",
    },
  },
  {
    comment: "I stayed here for one night and it was an unpleasant experience.",
    date: "2020-11-05T13:38:44.753Z",
    id: 2,
    rating: 4,
    user: {
      avatar_url: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
      id: 19,
      is_pro: false,
      name: "Christina",
    },
  },
];

export const commentsAdaptedToCLient = [
  {
    id: 3,
    date: "2020-11-05T13:38:44.753Z",
    src: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg",
    name: "Isaac",
    raiting: 5,
    text: "The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.",
    dateFormated: "November 2020",
  },
  {
    id: 2,
    date: "2020-11-05T13:38:44.753Z",
    src: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
    name: "Christina",
    raiting: 4,
    text: "I stayed here for one night and it was an unpleasant experience.",
    dateFormated: "November 2020",
  },
];

export const favoriteOffers = {
  "Amsterdam": [
    {
      city: {
        location: [50.938361, 6.959974],
        name: `Cologne`,
        zoom: 13,
      },
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      type: `house`,
      price: 493,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      id: 1,
      src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      isPremium: false,
      name: `The house among olive `,
      raiting: 3.8,
      isBookmark: false,
      coords: [50.957361, 6.9509739999999995],
      details: {
        bedrooms: 3,
        adults: 4,
        insideItems: [`Laptop friendly workspace`, `Washer`],
        description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      },
      user: {
        src: `img/avatar-angelina.jpg`,
        id: 25,
        name: `Angelina`,
        isSuper: true,
      },
      typeComponent: `main`,
    },
    {
      city: {
        location: [50.938361, 6.959974],
        name: `Cologne`,
        zoom: 13,
      },
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      type: `house`,
      price: 493,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      id: 2,
      src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      isPremium: false,
      name: `The house among olive `,
      raiting: 3.8,
      isBookmark: false,
      coords: [50.957361, 6.9509739999999995],
      details: {
        bedrooms: 3,
        adults: 4,
        insideItems: [`Laptop friendly workspace`, `Washer`],
        description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      },
      user: {
        src: `img/avatar-angelina.jpg`,
        id: 25,
        name: `Angelina`,
        isSuper: true,
      },
      typeComponent: `main`,
    },
  ],
};
