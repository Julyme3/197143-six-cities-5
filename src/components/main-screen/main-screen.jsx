import React, {useState} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {setOffersAction} from "../../store/actions";
import {OffersType} from "../../types";
import cities from "../../mocks/cities";
import OfferListMain from "../offer-list-main/offer-list-main";
import Map from "../map/map";
import ListCities from "../list-cities/list-cities";
import Sort from "../sort/sort";
import {sortOffers, getOffersByCity} from "../../offers";
import {SortType} from "../../const";
import MainLayout from "../../layouts/main-layout/main-layout";
import MainScreenEmpty from "../main-screen-empty/main-screen-empty";

const MainPage = (props) => {
  const [activeSortType, setSortType] = useState(SortType.POPULAR);
  const {offers, activeCity, defaultOffers} = props;
  const defaultOffersByCity = getOffersByCity(defaultOffers, activeCity);
  const sortedOffers = sortOffers(activeSortType, defaultOffersByCity);

  return (
    <MainLayout
      className="page--gray page--main"
      classNameWrap={`page__main--index ${offers.length ? `` : `page__main--index-empty`}`}
    >
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <ListCities
          cities={cities}
        />
      </div>
      <div className="cities">
        {offers.length &&
          <div className="cities__places-container container">
            <section className="cities__places places">
              <h2 className="visually-hidden">Places</h2>
              <b className="places__found">{offers.length} places to stay in {activeCity}</b>
              <Sort
                offers={offers}
                onClickSort={(sort) => setSortType(sort)}
                activeSortType={activeSortType}
              />
              <OfferListMain
                offers={sortedOffers}
                onChangeActiveOffer={props.onChangeActiveItem}
              />
            </section>
            <div className="cities__right-section">
              <section className="cities__map map">
                <Map
                  offers={offers}
                  width={`512px`}
                  height={`752px`}
                  activeCardId={props.activeItem}
                />
              </section>
            </div>
          </div>
          ||
          <MainScreenEmpty activeCity={activeCity} />}
      </div>
    </MainLayout>
  );
};

MainPage.propTypes = {
  offers: OffersType,
  defaultOffers: OffersType,
  setOffers: PropTypes.func.isRequired,
  activeCity: PropTypes.string.isRequired,
  onChangeActiveItem: PropTypes.func.isRequired,
  activeItem: PropTypes.any,
};

const mapStateToProps = (state) => ({
  offers: state.DATA.offers,
  defaultOffers: state.DATA.defaultOffers,
  activeCity: state.PROCESS.activeCity,
});

const mapDispatchToProps = (dispatch) => ({
  setOffers(selectedCity) {
    dispatch(setOffersAction(selectedCity));
  },
});

export {MainPage};
export default connect(mapStateToProps, mapDispatchToProps)(MainPage);
