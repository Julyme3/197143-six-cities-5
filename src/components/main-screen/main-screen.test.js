import React from "react";
import renderer from "react-test-renderer";
import {MainScreen} from "./main-screen";
import {offersAdapteToClient as offers} from "../../__mocks__/data/offer";

jest.mock(`../../layouts/main-layout/main-layout`, () => `MainLayout`);
jest.mock(`../main-screen-empty/main-screen-empty`, () => `MainScreenEmpty`);
jest.mock(`../list-cities/list-cities`, () => `ListCities`);
jest.mock(`../offer-list-main/offer-list-main`, () => `OfferListMain`);
jest.mock(`../sort/sort`, () => `Sort`);
jest.mock(`../map/map`, () => `Map`);

describe(`MainScreen render`, () => {
  it(`MainScreen empty render`, () => {
    const tree = renderer
      .create(
          <MainScreen
            offers={[]}
            activeCity={`Amsterdam`}
            onChangeActiveItem={()=>{}}
            activeItem={null}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`MainScreen with offers render`, () => {
    const tree = renderer
      .create(
          <MainScreen
            offers={offers}
            activeCity={`Amsterdam`}
            onChangeActiveItem={()=>{}}
            activeItem={null}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
