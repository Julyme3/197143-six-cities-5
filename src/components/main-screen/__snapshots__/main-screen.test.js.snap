// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen render MainScreen empty render 1`] = `
<MainLayout
  className="page--gray page--main"
  classNameWrap="page__main--index page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <ListCities
      cities={
        Array [
          "Amsterdam",
          "Cologne",
          "Brussels",
          "Paris",
          "Hamburg",
          "Dusseldorf",
        ]
      }
    />
  </div>
  <div
    className="cities"
  >
    <MainScreenEmpty
      activeCity="Amsterdam"
    />
  </div>
</MainLayout>
`;

exports[`MainScreen render MainScreen with offers render 1`] = `
<MainLayout
  className="page--gray page--main"
  classNameWrap="page__main--index "
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <ListCities
      cities={
        Array [
          "Amsterdam",
          "Cologne",
          "Brussels",
          "Paris",
          "Hamburg",
          "Dusseldorf",
        ]
      }
    />
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           places to stay in 
          Amsterdam
        </b>
        <Sort
          activeSortType="Popular"
          onClickSort={[Function]}
        />
        <OfferListMain
          offers={
            Array [
              Object {
                "city": Object {
                  "location": Array [
                    50.938361,
                    6.959974,
                  ],
                  "name": "Cologne",
                  "zoom": 13,
                },
                "coords": Array [
                  50.957361,
                  6.9509739999999995,
                ],
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "details": Object {
                  "adults": 4,
                  "bedrooms": 3,
                  "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                  "insideItems": Array [
                    "Laptop friendly workspace",
                    "Washer",
                  ],
                },
                "id": 1,
                "isBookmark": false,
                "isPremium": false,
                "name": "The house among olive ",
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "price": 493,
                "raiting": 3.8,
                "src": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                ],
                "type": "house",
                "typeComponent": "main",
                "user": Object {
                  "id": 25,
                  "isSuper": true,
                  "name": "Angelina",
                  "src": "img/avatar-angelina.jpg",
                },
              },
              Object {
                "city": Object {
                  "location": Array [
                    50.938361,
                    6.959974,
                  ],
                  "name": "Cologne",
                  "zoom": 13,
                },
                "coords": Array [
                  50.957361,
                  6.9509739999999995,
                ],
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "details": Object {
                  "adults": 4,
                  "bedrooms": 3,
                  "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                  "insideItems": Array [
                    "Laptop friendly workspace",
                    "Washer",
                  ],
                },
                "id": 2,
                "isBookmark": false,
                "isPremium": false,
                "name": "The house among olive ",
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "price": 493,
                "raiting": 3.8,
                "src": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                ],
                "type": "house",
                "typeComponent": "main",
                "user": Object {
                  "id": 25,
                  "isSuper": true,
                  "name": "Angelina",
                  "src": "img/avatar-angelina.jpg",
                },
              },
            ]
          }
          onChangeActiveOffer={[Function]}
        />
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <Map
            activeCardId={null}
            cityCoords={
              Array [
                50.938361,
                6.959974,
              ]
            }
            height="752px"
            offers={
              Array [
                Object {
                  "city": Object {
                    "location": Array [
                      50.938361,
                      6.959974,
                    ],
                    "name": "Cologne",
                    "zoom": 13,
                  },
                  "coords": Array [
                    50.957361,
                    6.9509739999999995,
                  ],
                  "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                  "details": Object {
                    "adults": 4,
                    "bedrooms": 3,
                    "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                    "insideItems": Array [
                      "Laptop friendly workspace",
                      "Washer",
                    ],
                  },
                  "id": 1,
                  "isBookmark": false,
                  "isPremium": false,
                  "name": "The house among olive ",
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "price": 493,
                  "raiting": 3.8,
                  "src": Array [
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  ],
                  "type": "house",
                  "typeComponent": "main",
                  "user": Object {
                    "id": 25,
                    "isSuper": true,
                    "name": "Angelina",
                    "src": "img/avatar-angelina.jpg",
                  },
                },
                Object {
                  "city": Object {
                    "location": Array [
                      50.938361,
                      6.959974,
                    ],
                    "name": "Cologne",
                    "zoom": 13,
                  },
                  "coords": Array [
                    50.957361,
                    6.9509739999999995,
                  ],
                  "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                  "details": Object {
                    "adults": 4,
                    "bedrooms": 3,
                    "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                    "insideItems": Array [
                      "Laptop friendly workspace",
                      "Washer",
                    ],
                  },
                  "id": 2,
                  "isBookmark": false,
                  "isPremium": false,
                  "name": "The house among olive ",
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "price": 493,
                  "raiting": 3.8,
                  "src": Array [
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  ],
                  "type": "house",
                  "typeComponent": "main",
                  "user": Object {
                    "id": 25,
                    "isSuper": true,
                    "name": "Angelina",
                    "src": "img/avatar-angelina.jpg",
                  },
                },
              ]
            }
            width="512px"
            zoom={13}
          />
        </section>
      </div>
    </div>
  </div>
</MainLayout>
`;
